File: Hackerrank_30daysOFcode_C#/day15.cs
using System;
class Node
{
	public int data;
	public Node next;
    public Node(int d){
        data=d;
        next=null;
    }

}
class Solution {

    public static Node insert(Node head,int data)
	{
        if(head == null){
            return new Node(data);
        }
        else if(head.next == null){
            head.next = new Node(data);
        }
        else{
            insert(head.next, data);
        }

        return head;
    }

    public static void display(Node head)
	{
		Node start=head;
		while(start!=null)
		{
			Console.Write(start.data+" ");
			start=start.next;
		}
	}
    static void Main(String[] args) {

		Node head=null;
        int T=Int32.Parse(Console.ReadLine());
        while(T-->0){
            int data=Int32.Parse(Console.ReadLine());
            head=insert(head,data);
        }
		display(head);
	}
}

File: 30daysOFcode_Go/day_15.go
package main

type Node struct {
	data interface{}
	next *Node
}

type LinkedList struct {
	head *Node
	tail *Node
	size int
}

func(list *LinkedList) Add(item interface{}) {
	newElement := &Node{data:item}
	if list.size == 0{
		list.head = newElement
		list.tail = newElement
	} else {
		list.tail.next = newElement
		list.tail = newElement
	}
	list.size++
}

func New() *LinkedList {
	return &LinkedList{}
}

func main() {
	a := New()
	for i:=0; i<5; i++ {
		a.Add(i)
	}
}